Src/BootCamp.Chapter/CredentialsManager.cs

```diff
@@ -0,0 +1,131 @@
+            }
+        }
+
+        public bool Login(Credentials credentials)
+        {
+            if (AreCredentialsValid(credentials))
+            {
+                if (DoesUserExist(credentials.Username))
+                {
+                    Console.WriteLine("Users Exists, Login is correct!");
+                    return UserCredentialsMatch(credentials);
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public void Register(Credentials credentials)
+        {
+            try
+            {
+                if (!DoesUserExist(credentials.Username))
+                {
+                    using (var writer = File.AppendText(_credentialsFile))
+                    {
+                        writer.WriteLine($"{credentials.Username},{credentials.Password}");
+                        Console.WriteLine($"Registration was successful, {credentials.Username}!");
+                    }
+                }
+                else
+                {
+                    throw new UserExistsException();
+                }
+            }
+            catch(UserExistsException msg)
+            {
+                Console.WriteLine(msg);
+            }  
+        }
+
+        private bool UserCredentialsMatch(Credentials credentials)
+        {
+            using (var reader = File.OpenText(_credentialsFile))
+            {
+                string unique_credentials = reader.ReadLine();
+                while (unique_credentials != null)
+                {
+                    bool usernameCheck = unique_credentials.Split(",")[0] == credentials.Username;
+                    bool passwordCheck = unique_credentials.Split(",")[1].Replace(" ", "") == credentials.Password;
+                    if (usernameCheck && passwordCheck)
+                    {
+                        Console.WriteLine($"Credentials Match! Welcome back {credentials.Username}!");
+                        return true;
+                    }
+                    else if (unique_credentials == null)
+                    {
+                        return false;
+                    }
+                    else
+                    {
+                        unique_credentials = reader.ReadLine();
+                    }
+                }
+                return false;
+            }
+        }
+
+        private bool DoesUserExist(string username)
+        {
+            using (var reader = File.OpenText(_credentialsFile))
+            {
+                string unique_credentials = reader.ReadLine();
+                while (unique_credentials != null)
+                {
+                    if (unique_credentials.Split(",")[0] == username)
+                    { 
+                        return true;
+                    }
+                    else if (unique_credentials == null)
```

Comment:
Codacy found an issue: [Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5768165)